src/shell.js

```diff
@@ -97,6 +97,10 @@ if (Module['ENVIRONMENT']) {
 var _scriptDir = import.meta.url;
 #else
 var _scriptDir = (typeof document !== 'undefined' && document.currentScript) ? document.currentScript.src : undefined;
+
+if (ENVIRONMENT_IS_NODE) {
```

Comment:
what's the difference between ENVIRONMENT_IS_NODE and ENVIRONMENT_HAS_NODE?

Reply:
There are environments like Electron that are not node, but include node functionality like require etc. That's unchanged in this PR, but Node workers become another such environment (they are ENVIRONMENT_IS_PTHREAD that HAS_NODE).